crates/biome_js_analyze/src/aria_analyzers/nursery/use_aria_activedescendant_tabindex.rs

```diff
@@ -0,0 +1,91 @@
+use crate::aria_services::Aria;
+use biome_analyze::{context::RuleContext, declare_rule, Rule, RuleDiagnostic};
+use biome_console::markup;
+use biome_js_syntax::jsx_ext::AnyJsxElement;
+use biome_rowan::AstNode;
+
+declare_rule! {
+    /// Enforce that `tabIndex` is assigned to non-interactive HTML elements with `aria-activedescendant`.
+    ///
+    /// aria-activedescendant is used to manage focus within a [composite widget].
+    /// The element with the attribute aria-activedescendant retains the active document focus;
+    /// it indicates which of its child elements has secondary focus by assigning the ID of that
+    /// element to the value of aria-activedescendant. This pattern is used to build a widget
+    /// like a search typeahead select list. The search input box retains document focus
+    /// so that the user can type in the input. If the down arrow key is pressed and
+    /// a search suggestion is highlighted, the ID of the suggestion element will be applied
+    /// as the value of aria-activedescendant on the input element.
+    ///
+    /// Because an element with aria-activedescendant must be tabbable,
+    /// it must either have an inherent tabIndex of zero or declare a tabIndex attribute.
```

Comment:
```suggestion
    /// Enforce that `tabIndex` is assigned to non-interactive HTML elements with `aria-activedescendant`.
    ///
    /// `aria-activedescendant` is used to manage to focus within a [composite widget].
    /// The element with the attribute `aria-activedescendant` retains the active document focus.
    ///
    /// It indicates which of its child elements has a secondary focus by assigning the ID of that
    /// element to the value of `aria-activedescendant`. This pattern is used to build a widget
    /// like a search typeahead select list. The search input box retains document focus
    /// so that the user can type in the input. If the down arrow key is pressed and
    /// a search suggestion is highlighted, the ID of the suggestion element will be applied
    /// as the value of `aria-activedescendant` on the input element.
    ///
    /// Because an element with `aria-activedescendant` must be tabbable,
    /// it must either have an inherent tabIndex of zero or declare a tabIndex attribute.
```